src/utils/internal.ts

```diff
@@ -1891,3 +1898,75 @@ export function areSameClaims(
 
   return ClaimType[type] === claim.type;
 }
+
+/**
+ * @hidden
+ */
+export function throwIfPendingAuthorizationExists(params: {
+  authorizationRequests: AuthorizationRequest[];
+  message: string;
+  authorization: Partial<Authorization>;
+  issuer?: Identity;
+  target?: string | Identity;
+}): void {
+  const {
+    authorizationRequests,
+    message,
+    authorization,
+    target: targetToCheck,
+    issuer: issuerToCheck,
+  } = params;
+
+  if (authorizationRequests.length === 0) {
+    return;
+  }
+
+  const hasPendingAuth = !!authorizationRequests.find(authorizationRequest => {
+    const { issuer, target, data } = authorizationRequest;
+
+    if (authorizationRequest.isExpired()) {
```

Comment:
FWIW I think sonar cloud is marking these overly aggresively. It gives +2 for each if due to nesting. I'd be OK ignoring complexity for this function.